codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/EventSourcedEntitySourceGenerator.scala

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.codegen.scalasdk.impl
+
+import com.akkaserverless.codegen.scalasdk.File
+import com.lightbend.akkasls.codegen.ModelBuilder
+import com.lightbend.akkasls.codegen.Format
+
+object EventSourcedEntitySourceGenerator {
+  import com.lightbend.akkasls.codegen.SourceGeneratorUtils._
+
+  def generateUnmanaged(
+      eventSourcedEntity: ModelBuilder.EventSourcedEntity,
+      service: ModelBuilder.EntityService): Seq[File] =
+    Seq(generateImplementationSkeleton(eventSourcedEntity, service))
+
+  def generateManaged(
+      eventSourcedEntity: ModelBuilder.EventSourcedEntity,
+      service: ModelBuilder.EntityService): Seq[File] =
+    Seq(
+      abstractEntity(eventSourcedEntity, service),
+      handler(eventSourcedEntity, service),
+      provider(eventSourcedEntity, service))
+
+  private[codegen] def abstractEntity(
+      eventSourcedEntity: ModelBuilder.EventSourcedEntity,
+      service: ModelBuilder.EntityService): File = {
+    val stateType = eventSourcedEntity.state.fqn.name
+    val abstractEntityName = eventSourcedEntity.abstractEntityName
+
+    implicit val imports =
+      generateImports(
+        Seq(eventSourcedEntity.state.fqn) ++
+        service.commands.map(_.inputType) ++
+        service.commands.map(_.outputType),
+        eventSourcedEntity.fqn.parent.scalaPackage,
+        otherImports = Seq("com.akkaserverless.scalasdk.eventsourcedentity.EventSourcedEntity"),
+        packageImports = Seq(service.fqn.parent.scalaPackage),
+        semi = false)
+
+    val commandHandlers = service.commands
+      .map { cmd =>
+        val methodName = cmd.name
+
+        val inputType = typeName(cmd.inputType)
+        val outputType = typeName(cmd.outputType)
+
+        s"""|def ${lowerFirst(methodName)}(currentState: $stateType, ${lowerFirst(cmd.inputType.name)}: $inputType): EventSourcedEntity.Effect[$outputType]
+            |""".stripMargin
+
+      }
+
+    val eventHandlers =
+      eventSourcedEntity match {
+        case ModelBuilder.EventSourcedEntity(_, _, _, events) =>
+          events.map { event =>
+            s"""|def ${lowerFirst(event.fqn.name)}(currentState: ${typeName(
+              eventSourcedEntity.state.fqn)}, ${lowerFirst(event.fqn.name)}: ${typeName(event.fqn)}): ${typeName(
```

Comment:
We can use `$stateType` instead of `${typeName(eventSourcedEntity.state.fqn)}`. Will make is shorter to read.